server/common/rest.py

```diff
@@ -40,6 +42,43 @@
 from server.common.utils.uns import spatial_metadata_get
 from server.dataset import dataset_metadata
 
+LOCAL_DEV_USER_PREFIX = "test-user-"
+RDEV_USER_ID = "rdev-user"
+
+
+HEADER_USER_ID_CANDIDATES = ()
+
+
+def _get_request_user_id(req) -> Optional[str]:
+
+    if req is None:
+        return None
+
+    user = req.headers.get(
+        "X-Auth-Request-User",
+    )
+    if user:
+        return user
+
+
+def _resolve_request_user_id(req) -> Optional[str]:
```

Comment:
Central _resolve_request_user_id: respects disable_auth, returns a dev or rdev fallback where appropriate, and otherwise requires an authenticated user.